src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java

```diff
@@ -178,11 +185,14 @@ public void checkExistence() {
             // Handle response if valid itineraries exist.
             if (plan != null && plan.itineraries != null) {
                 for (Itinerary itineraryCandidate : plan.itineraries) {
-                    // If a matching itinerary is found, save the date with the matching itinerary.
-                    // The matching itinerary will replace the original trip.itinerary.
-                    if (ItineraryUtils.itinerariesMatch(referenceItinerary, itineraryCandidate)) {
-                        result.handleValidDate(otpRequest.date, itineraryCandidate);
-                        hasMatchingItinerary = true;
+                    // Make sure itinerary is same day as request date
+                    if (ItineraryUtils.isSameDay(itineraryCandidate, otpRequest.date, tripIsArriveBy)) {
```

Comment:
Could you change `OtpRequest#date` to `OtpRequest#requestDateTime`? I'm finding its current name kind of confusing (probably originally my mistake!).

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/102/commits/911c618356112c4db4b089e158aab92ab6c23062.